register-computer/main.py

```diff
@@ -102,46 +102,61 @@ def __connect_to_activedirectory():
 def __generate_password(length=40):
     return str(uuid.uuid4()) + "-" + str(uuid.uuid4())
 
-def __shorten_computer_name(computer_name, gce_instance):    
-    # Look for MIG related metadata
-    has_metadata = ("metadata" in gce_instance.keys() and
-                    "items" in gce_instance["metadata"])
-    if has_metadata:
-      metadata_created_by = next((x for x in gce_instance["metadata"]["items"] if x["key"] == "created-by"), None)
-
-    is_mig = (has_metadata and 
-              metadata_created_by and              
-              "instanceGroupManagers" in metadata_created_by["value"])
-    
-    if not is_mig:
-      return computer_name
-    
-    if ("labels" in gce_instance.keys() and
-        'goog-gke-node' in gce_instance["labels"]):
-      # Generate GKE node naming convention k-XXXXXXXX-YYYY 
-      # (k - Kubernetes node
-      #  X - unique value given to the cluster's node pool
-      #  Y - unique value given to each instance by the MIG
-      instance_name_parts = computer_name.rsplit('-',2)
-      node_pool_hash = instance_name_parts[-2]
-      unique_id = instance_name_parts[-1]
-      new_computer_name = 'k-' + node_pool_hash + '-' + unique_id
-    else: 
-      # Generate MIG naming convention XXXXX-YYYY-ZZZZ 
-      # (X - partial MIG name
-      #  Y - hashed value of MIG name
-      #  Z - unique value given to each instance by the MIG
-      instance_name_parts = computer_name.rsplit('-',1)
-      mig_name = instance_name_parts[-2]
-      unique_id = instance_name_parts[-1]
-      # Create a hash that produces 4 hex characters
-      hasher = blake2b(digest_size=2)
-      hasher.update(mig_name.encode("utf-8"))
-      mig_name_hash = hasher.hexdigest()
-      # Get first 5 characters from MIG's name
-      mig_name_prefix = mig_name[:5]
-      new_computer_name = mig_name_prefix + '-' + mig_name_hash + '-' + unique_id
-    return new_computer_name
+def __get_managed_instance_group_for_instance(gce_instance):
+    if ("metadata" in gce_instance.keys() and "items" in gce_instance["metadata"]):
+        metadata_created_by = next((x for x in gce_instance["metadata"]["items"] if x["key"] == "created-by"), None)
+
+    if (metadata_created_by and "instanceGroupManagers" in metadata_created_by["value"]):
+        # https://cloud.google.com/compute/docs/instance-groups/getting-info-about-migs#checking_if_a_vm_instance_is_part_of_a_mig
+        # The "created-by" metadata value is in the format
+        # projects/[number]/zones/[zone]/instanceGroupManagers/[mig-name]
+        # Return only the last part (the MIG name) from the value
+        return (metadata_created_by["value"]).rsplit('/',1)[-1]
```

Comment:
Add space after comma

(in other places as well)

Reply:
Fixed